CuBeanzQueryable.sol

```diff
@@ -0,0 +1,251 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+error NotOwner();
+error MaxTeamMints();
+error RoundSoldOut();
+error SaleNotStarted();
+error InvalidValue();
+error MaxMints();
+error ContractMint();
+/*
+
+░█████╗░██╗░░░██╗██████╗░███████╗░█████╗░███╗░░██╗███████╗
+██╔══██╗██║░░░██║██╔══██╗██╔════╝██╔══██╗████╗░██║╚════██║
+██║░░╚═╝██║░░░██║██████╦╝█████╗░░███████║██╔██╗██║░░███╔═╝
+██║░░██╗██║░░░██║██╔══██╗██╔══╝░░██╔══██║██║╚████║██╔══╝░░
+╚█████╔╝╚██████╔╝██████╦╝███████╗██║░░██║██║░╚███║███████╗
+░╚════╝░░╚═════╝░╚═════╝░╚══════╝╚═╝░░╚═╝╚═╝░░╚══╝╚══════╝
+
+Dev: Shimazu Bohoro
+*/
+
+
+import "erc721a/contracts/extensions/ERC721AQueryable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+
+
+contract CuBeanzQueryable is  ERC721AQueryable , Ownable, ReentrancyGuard {
+    using Strings for uint256;
+    using ECDSA for bytes32;
+
+    //@dev mint constraints
+    uint16 public teamMints;
+    uint16 maxTeamMints = 444;
+    uint16  roundOneSupply = 2222;
+    uint16  roundTwoSupply = 1778;
+    uint16 public roundOneMints;
+    uint16 public roundTwoMints;
+
+    uint8 maxRoundOneMints = 2;
+    uint8 maxRoundTwoMints = 1;
+    uint8 public maxAttacks  = 1;
+    
+
+    uint80 public burnRate = 10000 ether;
+    uint80 public cubePrice = 22000 ether; 
+    
+
+    bool public roundOneLive;
+    bool public roundTwoLive;
+
+    bool burnMintActive;
+
+    bool public revealed;   
+    
+
+    
+    //@dev ryoToken obtained by burning cubes
+    IERC20 public ryoToken; 
+
+
+    
+
+
+    //@dev tokenUri factory
+    string public baseURI;
+    string public notRevealedUri;
+    string public uriSuffix = ".json";
+    
+   //@dev maps tokenIds to number of times they've attacked
+   //@notice supply will never be > uint16 max value(65535) therefore we can reference to tokenIds by uint16
+   //@notice numAttacks will never be more than 8 since maxAttacks will never be more than 8
+    mapping(uint16 => uint8) public numAttacks;
+
+
+
+    // @dev these mappings track how many one has minted on public and WL respectively
+    mapping(address =>uint8) public rOneMapping;
+    mapping(address => uint8) public rTwoMapping;
```

Comment:
Consider using `_getAux(msg.sender)` and `_setAux(msg.sender, value)` for round 1.

And to get the number of round 2 mints, use `_numberMinted(msg.sender) - _getAux(msg.sender)`.